### PR TITLE
More claim tests

### PR DESCRIPTION
**Pull request by g4s8**
#761 - added test for claim with one child and test for claim with many children

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#998](https://github.com/zerocracy/farm/issues/998) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#998](https://github.com/zerocracy/farm/issues/998) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1270783041); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/test/java/com/zerocracy/tk/project/TkClaimTest.java [hunk 1]
```diff
@@ -76,4 +79,80 @@ public void renderClaimWithNoChildrenXml() throws Exception {
...
+    }
+
+    @Test
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    public void renderClaimWithManyChildren() throws Exception {
+        final FtFarm farm = new FtFarm(new PropsFarm(new FkFarm()));
+        final long parent = 164L;
+        final int children = Tv.FIFTY;
+        final Project proj = farm.find("@id='C00000000'").iterator()
+            .next();
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/tk/project/TkClaimTest.java]: @g4s8 can we move this up to the previous line?

File: src/test/java/com/zerocracy/tk/project/TkClaimTest.java [hunk 2]
```diff
@@ -76,4 +79,80 @@ public void renderClaimWithNoChildrenXml() throws Exception {
...
+        new ClaimOut().type("test").cid(parent).postTo(proj);
+        final XML xml = new Claims(proj).iterate().iterator().next();
+        for (int number = 0; number < children; ++number) {
+            new ClaimIn(xml).copy().cid((long) (Tv.THOUSAND + number))
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/tk/project/TkClaimTest.java]: @g4s8 let's move `new ClaimIn` outside the loop. This should allow us to remove the PMD warning suppression above.

File: src/test/java/com/zerocracy/tk/project/TkClaimTest.java [hunk 3]
```diff
@@ -76,4 +79,80 @@ public void renderClaimWithNoChildrenXml() throws Exception {
...
+    @Test
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    public void renderClaimWithManyChildren() throws Exception {
+        final FtFarm farm = new FtFarm(new PropsFarm(new FkFarm()));
+        final long parent = 164L;
+        final int children = Tv.FIFTY;
+        final Project proj = farm.find("@id='C00000000'").iterator()
+            .next();
+        new ClaimOut().type("test").cid(parent).postTo(proj);
+        final XML xml = new Claims(proj).iterate().iterator().next();
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/tk/project/TkClaimTest.java]: @g4s8 looks like we're only using this once below, so we should be able to inline (after we move `ClaimIn` out of the loop, see below).

**Comment by carlosmiranda** [comment]
@yegor256 ping

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by yegor256** [comment]
@rultor merge

**Comment by carlosmiranda** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/998#issuecomment-389043790))

@carlosmiranda This is what I know about this job, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#998](https://github.com/zerocracy/farm/issues/998) is [in scope](http://www.zerocracy.com/policy.html#14) for 8days
  * The role is `REV`
  * The job is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) for [8days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will get 25% of their hourly rate for this job
  * Some equity will be [vested](http://www.zerocracy.com/policy.html#37) on completion
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @g4s8[/z](https://www.0crat.com/u/g4s8): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%23998%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by carlosmiranda** [comment]
@yegor256 can we move this out of scope? `0crat` seems to have missed it.

**Comment by carlosmiranda** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@0crat out

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/zerocracy/farm/issues/998#issuecomment-390187586))

@yegor256 The job [#998](https://github.com/zerocracy/farm/issues/998) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by ypshenychka** [comment]
@carlosmiranda According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Three issues were found during code review, one of which was cosmetic.
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by carlosmiranda** [comment]
@ypshenychka confirmed, thanks

**Comment by ypshenychka** [comment]
@carlosmiranda Thank you.

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@789b861f-9fb0-4ee3-acf5-4b943019aa6f, thread: BkParallel-8
com.zerocracy.radars.github.ReSafe[82] java.io.IOException: java.lang.IllegalStateException: Duplicate claims are not allowed in [C3NDPUA8L](https://www.0crat.com/p/C3NDPUA8L), can't add this XML to 2 existing ones:
<?xml version="1.0" encoding="UTF-8"?>
<claim id="1410665702">
   <created>2018-05-21T06:38:21.157Z</created>
   <type>Complete `QA` review</type>
   <params>
      <param name="quality">acceptable</param>
      <param name="job">gh:zerocracy/farm#904</param>
      <param name="repo">zerocracy/farm</param>
      <param name="issue">904</param>
      <param name="comment">390566606</param>
   </params>
   <token>github;zerocracy/farm;904;390566606</token>
   <author>ypshenychka</author>
</claim>

com.zerocracy.pm.Claims[112] java.lang.IllegalStateException: Duplicate claims are not allowed in [C3NDPUA8L](https://www.0crat.com/p/C3NDPUA8L), can't add this XML to 2 existing ones:
<?xml version="1.0" encoding="UTF-8"?>
<claim id="1410665702">
   <created>2018-05-21T06:38:21.157Z</created>
   <type>Complete `QA` review</type>
   <params>
      <param name="quality">acceptable</param>
      <param name="job">gh:zerocracy/farm#904</param>
      <param name="repo">zerocracy/farm</param>
      <param name="issue">904</param>
      <param name="comment">390566606</param>
   </params>
   <token>github;zerocracy/farm;904;390566606</token>
   <author>ypshenychka</author>
</claim>

```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm): Issue: zerocracy/farm#998, Action: `created`

**Pull request merged**